test/checks/design/InterfaceIsTypeCheckTest.hx

```diff
@@ -0,0 +1,40 @@
+package checks.design;
+
+import checkstyle.checks.design.InterfaceIsTypeCheck;
+
+class InterfaceIsTypeCheckTest extends CheckTestCase<InterfaceIsTypeCheckTests> {
+
+	public function testWithJustVars() {
+		assertMsg(new InterfaceIsTypeCheck(), InterfaceIsTypeCheckTests.TEST1, "Interfaces should describe a type and hence have methods");
+	}
+
+	public function testMarkerInterface() {
+		assertNoMsg(new InterfaceIsTypeCheck(), InterfaceIsTypeCheckTests.TEST2);
+	}
+
+	public function testNoMarkerInterface() {
+		var chk = new InterfaceIsTypeCheck();
+		chk.allowMarkerInterfaces = false;
+		assertMsg(chk, InterfaceIsTypeCheckTests.TEST2, "Interfaces should describe a type and hence have methods");
```

Comment:
I like to allow properties with getters / setters? (default, null) only but not sure.

The only problem this solves is interfaces with just constants.